lib/controller/controller.go

```diff
@@ -156,6 +170,12 @@ const (
 	DefaultRetryPeriod = 2 * time.Second
 	// DefaultTermLimit is used when option function TermLimit is omitted
 	DefaultTermLimit = 30 * time.Second
+	// DefaultMetricsPort is used when option function MetricsPort is omitted
+	DefaultMetricsPort = 0
```

Comment:
Should there be a port other than random we should default to? Like some convention for all provisioners. Will the port need to be exposed via the pod yaml? Because then a convention would make it easy to distribute yamls with the port already filled in

Reply:
Though port `0` means picking a random port in network programming, I used it to [disabling metrics server](https://github.com/kubernetes-incubator/external-storage/pull/796/files/58b4e13e3d6da50dfde002534d411e2efd3c3963#diff-16daa5c3f02eee06add7f1af0a3fa901R552) here, like kubelet `--cadvisor-port` flag. Sorry, comments here is a bit confusing.

> Will the port need to be exposed via the pod yaml? Because then a convention would make it easy to distribute yamls with the port already filled in

It depends on provisioner implementation. Provisioner can have a non-zero metrics port and pass it to `NewProvisionController()` to enable metrics server by default.

My intent is to make this feature optional for backward compatibility, each provisioner can customize based on needs, e.g. disable/enable by default, choose default port. We can have some convention for default port (e.g. 8000), but we may need to add another option to toggle feature. 

What do you prefer? Do you want to enable metrics server by default?

Reply:
hmm okay I agree with you actually let's disable it by default